cmd/kata-local-ci

```diff
@@ -0,0 +1,260 @@
+    ssh root@$VM_IPADDR -o StrictHostKeyChecking=no -o UpdateHostKeys=no "$@"
+}
+
+while getopts fFhk:d:o:s:n:w:j: flag
+do
+    case "${flag}" in
+        h) help; exit 0 ;;
+        d) distro=$OPTARG;;
+        w) directory=$OPTARG;;
+        j) job=$OPTARG;;
+        v) name=$OPTARG;;
+        k) key=$OPTARG;;
+	o) disk=$OPTARG;;
+	s) size=$OPTARG;;
+	c) cpus=$OPTARG;;
+	m) memory=$OPTARG;;
+	n) network=$OPTARG;;
+	f) install=true ;;
+        F) revert=false;;
+        *) help; exit 1;;
+    esac
+done
+
+if [ ! -d "$kci" ]; then
+    mkdir "$kci"
+fi
+
+if [ ! -f "$kci/"$distro-config ]; then
+    install=true
+fi
+
+if [ -z "$disk" ]; then
+    disk=/var/lib/libvirt/images/kata-$distro-ci.qcow2
+fi
+
+if [ ! -f "$disk" ] ; then
+    install=true
+fi
+
+if [ ! -d "$directory"/kata-containers ]; then
+    echo "ERROR: The work directory $directory does not contain a 'kata-containers' subdirectory"
+    exit 4
+fi
+if [ ! -d "$directory"/tests ]; then
+    echo "ERROR: The work directory $directory does not contain a 'tests' subdirectory"
+    exit 4
+fi
+
+
+function vm_install() {
+    distro_commands=""
+    firstboot_cmds=(
+	--firstboot-command 'useradd -m  kata -s /bin/bash'
+	--firstboot-command 'echo "kata ALL=(ALL) NOPASSWD:ALL" >> /etc/sudoers.d/kata'
+	--firstboot-command "echo 'export PATH=\$PATH:/usr/local/go/bin' >> /home/kata/.bashrc"
+	--firstboot-command 'mkdir -p /home/kata/.ssh'
+	--firstboot-command "echo $(cat $key) > /home/kata/.ssh/authorized_keys"
+	--firstboot-command 'chown kata:kata -R /home/kata'
+	--firstboot-command 'chown kata:kata -R /home/kata'
+        --firstboot-command "swapoff -a"
+        --firstboot-command "sed -i -e 's/^.*swap.*$/#\0/g' /etc/fstab"
+    )
+    final_firstboot_cmd=()
+    ubuntu_distro_commands=(
+        --firstboot-command 'usermod -a -G sudo kata'
+	--update
+	--firstboot-command 'dhclient'
+	--firstboot-command 'systemctl enable serial-getty@ttyS0'
+	--firstboot-command 'systemctl start serial-getty@ttyS0'
+    )
+    rh_distro_commands=(
+        --selinux-relabel
+        --edit '/etc/selinux/config:s/SELINUX=enforcing/SELINUX=permissive/'
+        --firstboot-command 'systemctl stop firewalld'
+        --firstboot-command 'systemctl disable firewalld'
+    )
+    disable_cgroupv2=(
+        --firstboot-command 'grubby --update-kernel=ALL --args="systemd.unified_cgroup_hierarchy=0"'
+        --firstboot-command 'reboot'
+    )
+
+    case "$distro" in
+	ubuntu)
+	    distro_name=ubuntu-18.04
+	    distro_commands=("${ubuntu_distro_commands[@]}")
+	    osvariant=ubuntu18.04
+	    ;;
+	centos)
+	    distro_name=centos-8.2
+	    osvariant=centos8
+	    distro_commands=("${rh_distro_commands[@]}")
+	    ;;
+	fedora)
+	    distro_name=fedora-32
+	    osvariant=fedora32
+	    distro_commands=("${rh_distro_commands[@]}")
+	    final_firstboot_cmd=("${disable_cgroupv2[@]}")
+	    ;;
+	*)
+	    echo "Distro $1 not recognized. Please choose ubuntu, centos or fedora"
+            exit 3
```

Comment:
ooi, why all the different return values? These aren't documented in the usage, so isn't `0` or `1` sufficient?

Reply:
Computer scientists and their binary!

Reply:
Fixed